azure-kusto-data/azure/kusto/data/client.py

```diff
@@ -853,19 +853,61 @@ def execute_streaming_ingest(
 
         self._execute(endpoint, database, None, stream, self._streaming_ingest_default_timeout, properties)
 
-    def _execute(
-        self, endpoint: str, database: str, query: Optional[str], payload: Optional[io.IOBase], timeout: timedelta, properties: ClientRequestProperties = None
+    def execute_streaming_query(
+        self, database: str, query: str, timeout: timedelta = _KustoClientBase._query_default_timeout, properties: Optional[ClientRequestProperties] = None
+    ) -> Response:
+        """
+        Query directly from Kusto database using streaming output.</p>
+        This method queries the Kusto database into an asyncio stream.
+
+        example usage::
+
+            my_file = open("some_file", "wb")
+            with client.execute_streaming_query(...) as response:
+                for chunk in response.iter_content(1024):
+                    file.write(chunk)
+
+        :return: a Response that can be read as chunks
+        """
+        response = self._retrieve_response(self._query_endpoint, database, properties, query, None, timeout, stream=True)
+
+        try:
+            response.raise_for_status()
+        except Exception as e:
+            self._handle_http_error(e, self._query_endpoint, None, response, response.json(), response.text)
+
+        return response
+
+    def _retrieve_response(
+        self,
+        endpoint: str,
+        database: str,
+        properties: Optional[ClientRequestProperties],
+        query: str,
+        payload: Optional[io.IOBase],
+        timeout: timedelta,
+        stream: bool,

```

Comment:
Set a default

Reply:
Why?

It's a private function, so it won't help the user to set it, and for us it makes sure we don't accidentally miss the parameter

Reply:
It comes from the same place as the previous comment - we should have 1 _execute with default as most of the flows use stream=false